File: AirlinesAPI/app/Models/Flight.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\HasOne;

class Flight extends Model
{
    use HasFactory;

    public $timestamps = false;

    protected $fillable = [
        'id',
        'airline_id',
        'origin_id',
        'destination_id',
        'distance',
        'flight_time',
        'huf_per_km',
    ];

    protected $casts = [
        'distance' => 'integer',
        'flight_time' => 'integer',
        'huf_per_km' => 'integer',
    ];

    public function airline(): BelongsTo
    {
        return $this->belongsTo(Airline::class, 'airline_id');
    }

    public function originCity(): BelongsTo
    {
        return $this->belongsTo(City::class, 'origin_id');
    }

    public function destinationCity(): BelongsTo
    {
        return $this->belongsTo(City::class, 'destination_id');
    }
}

File: AirlinesAPI.Models/Flight.cs
using System;
using System.Text.Json.Serialization;

namespace AirlinesAPI.Models;

public sealed class Flight : IdModel, IEquatable<Flight> {
    [JsonPropertyName("airline_id"), JsonRequired]
    public int AirlineId { get; set; }

    [JsonPropertyName("origin_id"), JsonRequired]
    public int OriginId { get; set; }

    [JsonPropertyName("destination_id"), JsonRequired]
    public int DestinationId { get; set; }

    [JsonPropertyName("distance"), JsonRequired]
    public int Distance { get; set; }

    [JsonPropertyName("flight_time"), JsonRequired]
    public int FlightTime { get; set; }

    [JsonPropertyName("huf_per_km"), JsonRequired]
    public int HufPerKm { get; set; }

    public bool Equals(Flight? other) {
        return other is not null
            && Id            == other.Id
            && AirlineId     == other.AirlineId
            && OriginId      == other.OriginId
            && DestinationId == other.DestinationId
            && Distance      == other.Distance
            && FlightTime    == other.FlightTime
            && HufPerKm      == other.HufPerKm;
    }

    public override string ToString() => $"from {OriginId} to {DestinationId} ({Id})";
}
